pkg/process/checks/net.go

```diff
@@ -130,48 +130,48 @@ func (c *ConnectionsCheck) enrichConnections(conns []*model.Connection) []*model
 	return conns
 }
 
-func (c *ConnectionsCheck) diffTelemetry(tel *model.ConnectionsTelemetry) *model.CollectorConnectionsTelemetry {
+func (c *ConnectionsCheck) diffAndFormatTelemetry(tel map[string]int64) map[string]int64 {
 	if tel == nil {
 		return nil
 	}
 	// only save but do not report the first collected telemetry to prevent reporting full monotonic values.
 	if c.lastTelemetry == nil {
-		c.lastTelemetry = &model.CollectorConnectionsTelemetry{}
+		c.lastTelemetry = make(map[string]int64)
 		c.saveTelemetry(tel)
 		return nil
 	}
 
-	cct := &model.CollectorConnectionsTelemetry{
-		KprobesTriggered:          tel.MonotonicKprobesTriggered - c.lastTelemetry.KprobesTriggered,
-		KprobesMissed:             tel.MonotonicKprobesMissed - c.lastTelemetry.KprobesMissed,
-		ConntrackRegisters:        tel.MonotonicConntrackRegisters - c.lastTelemetry.ConntrackRegisters,
-		ConntrackRegistersDropped: tel.MonotonicConntrackRegistersDropped - c.lastTelemetry.ConntrackRegistersDropped,
-		DnsPacketsProcessed:       tel.MonotonicDnsPacketsProcessed - c.lastTelemetry.DnsPacketsProcessed,
-		ConnsClosed:               tel.MonotonicConnsClosed - c.lastTelemetry.ConnsClosed,
-		ConnsBpfMapSize:           tel.ConnsBpfMapSize,
-		UdpSendsProcessed:         tel.MonotonicUdpSendsProcessed - c.lastTelemetry.UdpSendsProcessed,
-		UdpSendsMissed:            tel.MonotonicUdpSendsMissed - c.lastTelemetry.UdpSendsMissed,
-		ConntrackSamplingPercent:  tel.ConntrackSamplingPercent,
-		DnsStatsDropped:           tel.DnsStatsDropped,
+	cct := map[string]int64{
+		"kprobes_triggered":           tel["MonotonicKprobesTriggered"] - c.lastTelemetry["KprobesTriggered"],
```

Comment:
Can we use constants for the map keys on the right here? This would prevent typos from causing no data to be reported. For example, `"MonotonicKprobesTriggered"` is used in 3 different places.